### PR TITLE
Add record operation for channel

### PR DESCRIPTION
**Pull request by vemarun**
Hi @adambezecny ,

Can you please review my code.
I have tried to implement recording functionality.

Thanks.

image

### CODE REVIEW

**Comment by adambezecny** [comment]
hi,

thank you for your contribution! It has been some time since I actively worked with asterisk. Can you provide additional information what this change will bring in terms of functionality? Also is there any quick way how to test it? Ideally is it possible to test it somehow against dockerized version of Asterisk? Can you provide example (see examples folder) showing this new feature in action?

**Comment by vemarun** [comment]
Hi,
This functionality will add on demand recording on a channel - based on any flag / variable value we can start the recording.
I have put example in example.rs
```
fn stasis_start(event: StasisStart) {
    tokio::spawn(async move {
        info!("stasis_start: {:#?}", event);
        debug!("Answering channel {} now!", &event.channel.id);
        ARICLIENT.answer(&event.channel.id).await.unwrap();
        debug!("Channel {} answered!", &event.channel.id);
        ARICLIENT.record(&event.channel.id, None, None).await.unwrap();    //  start recording after channel answer
        debug!("Channel {} recording started!", &event.channel.id);
    });
}
```
I tested it by connecting to a remote server from local.
By default it starts recording in /var/spool/asterisk/recording/<channel_id> with file format wav
image

**Comment by adambezecny** [comment]
ok, bear with me some time. I am trying to prepare some dockerized asterisk to test. I would also like to document it here to help other people quickly spin up local asterisk environment that can be used to test the client.

File: src/client.rs [hunk 1]
```diff
@@ -568,4 +568,44 @@ impl ChannelsAPI for AriClient {
         eval_status_code!(status, StatusCode::CREATED, Some(body_str));
         Ok(())
     }
+
+    async fn raise_dtmf(
```
Comment (adambezecny) [inline src/client.rs]: still didn't have chance to test previous change. please for now rollback this commit. should be addressed separately. i will try to test recording this week.

Comment (vemarun) [reply src/client.rs]: Sorry for that, rolled back 1 commit,
moved dtmf to another separate temp branch. Will create another PR for dtmf after this merge.
Thanks for giving time.

**Comment by adambezecny** [comment]
hi,

I was able to make dockerized version of asterisk, see folder https://github.com/jabber-tools/asterisk-ari-client-rs/tree/main/examples/ast-docker and test your change. I have published new version 0.1.2. Thanks again for the contribution. Please go ahead and submit the other change as discussed. 

In general I am thinking about couple of more changes (since I already touched the code after quite a long time): 

-updating dependencies like tokio, etc. to latest available versions
-getting rid of async-trait crate since rust now supports asynchronous traits natively. 
-bringing more APIs, maybe even aligning with what is available in go version (see https://docs.asterisk.org/Configuration/Interfaces/Asterisk-REST-Interface-ARI/ARI-Libraries/ library **asterisk-ari-go**)
- this would allow to promote library more and maybe include it into this page as official rust library for ARI

What are your plans for future extensions of this library? Can you maybe drop me some details about your use case, features you want to support in the future, etc?

thanks,

Adam

**Pull request closed without merging**